c2rust/src/main.rs

```diff
@@ -29,12 +28,11 @@ impl SubCommand {
     /// They are of the form `c2rust-{name}`.
     pub fn find_all() -> anyhow::Result<Vec<Self>> {
         let c2rust = env::current_exe()?;
-        let c2rust_name: &Utf8Path = c2rust
+        let c2rust_name = c2rust
             .file_name()
-            .map(Path::new)
-            .ok_or_else(|| anyhow!("no file name: {}", c2rust.display()))?
-            .try_into()?;
-        let c2rust_name = c2rust_name.as_str();
+            .ok_or_else(|| anyhow!("no file name for c2rust: {}", c2rust.display()))?
```

Comment:
Is there a reason you didn't use `.with_context()` here?

Reply:
Ah, I didn't know `Option` `impl`s `Context`.  I can change that.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/644/commits/0a449f71e10952eb0f02822e1cc2ee04ed1806b3.

Reply:
Actually, I reverted that.  There are tons of `.ok_or_else(|| anyhow!(...))`s everywhere, so I think if we want to change it, we should just change it everywhere.